### PR TITLE
feat(gatsby-source-shopify): Filter fetched collections

### LINKED ISSUE #14952: Error in gatsby-source-shopify
## Description

I am getting the following error during `npm run develop`:

```
gatsby-source-shopify/<shop-name> starting to fetch data from Shopify
error
error an error occured while sourcing data
error - 
  message:   access denied
  locations: 
    - 
      line:   4
      column: 7
  path: 
    - shop
    - blogs
error query: 
  """

      query GetBlogs($first: Int!, $after: String) {
        shop {
          blogs(first: $first, after: $after) {
            pageInfo {
              hasNextPage
            }
            edges {
              cursor
              node {
                id
                title
                url
              }
            }
          }
        }
      }

  """
variables: 
  first: 250
  after: null
```

### Steps to reproduce

Create a Storefront API Access Token that doesn't have access to "Read content like articles, blogs, and comments" and try to run gatsby

### Expected result

Site builds with only the queries I provide and successfully builds.

### Actual result

Unknown query is running and erroring. No where in my site am I running this query so I wouldn't expect it to run. My site doesn't need blogs from Shopify. The error is expected, the fact that it is running is not.

### Environment

  System:
    OS: Linux 4.15 Ubuntu 18.04.1 LTS (Bionic Beaver)
    CPU: (8) x64 AMD EPYC 3251 8-Core Processor
    Shell: 4.4.19 - /bin/bash
  Binaries:
    Node: 12.2.0 - ~/.nvm/versions/node/v12.2.0/bin/node
    npm: 6.9.0 - ~/.nvm/versions/node/v12.2.0/bin/npm
  npmPackages:
    gatsby: ^2.9.4 => 2.9.4
    gatsby-image: ^2.1.4 => 2.1.4
    gatsby-plugin-manifest: ^2.1.1 => 2.1.1
    gatsby-plugin-offline: ^2.1.3 => 2.1.3
    gatsby-plugin-react-helmet: ^3.0.12 => 3.0.12
    gatsby-plugin-sharp: ^2.1.5 => 2.1.5
    gatsby-source-filesystem: ^2.0.39 => 2.0.39
    gatsby-source-graphql: ^2.0.19 => 2.0.19
    gatsby-source-shopify: ^2.0.32 => 2.0.32
    gatsby-transformer-sharp: ^2.1.21 => 2.1.21
  npmGlobalPackages:
    gatsby-cli: 2.6.7

### PR DESCRIPTION
**Pull request by martingtheodo**
## Description

In gatsby-source-shopify, add a plugin option to fetch only some of the available objects. (instead of having access errors trying to fetch them)

## Related Issues

Resolves #14952

### CODE REVIEW

File: packages/gatsby-source-shopify/src/gatsby-node.js [hunk 1]
```diff
@@ -63,28 +78,53 @@ export const sourceNodes = async (
...
-            createNode(await ProductVariantNode(imageArgs)(edge.node))
-          )
-
-        if (x.options)
-          await forEach(x.options, async option =>
-            createNode(await ProductOptionNode(imageArgs)(option))
-          )
-      }),
-      createShopPolicies(args),
+      ...(fetchCollections.includes(ARTICLE)
```
Comment (freiksenet) [inline packages/gatsby-source-shopify/src/gatsby-node.js]: Those inline ternaries are a bit hard to read. I wonder if you could separate it into normal if/else blocks? One could push promises to a list and then await for that list in the end.

File: packages/gatsby-source-shopify/src/gatsby-node.js [hunk 2]
```diff
@@ -26,7 +35,13 @@ import {
 
 export const sourceNodes = async (
   { actions: { createNode, touchNode }, createNodeId, store, cache, reporter },
-  { shopName, accessToken, verbose = true, paginationSize = 250 }
+  {
+    shopName,
+    accessToken,
+    verbose = true,
+    paginationSize = 250,
+    fetchCollections = ALL_NODES,
```
Comment (freiksenet) [inline packages/gatsby-source-shopify/src/gatsby-node.js]: Could you update the README to document that option?

**Comment by martingtheodo** [comment]
Thanks for your comments, they all make sense.
I will try to integrate them next week.

File: packages/gatsby-source-shopify/src/nodes.js [hunk 3]
```diff
@@ -7,16 +7,33 @@ import { createRemoteFileNode } from "gatsby-source-filesystem"
...
-const BLOG = `Blog`
-const COLLECTION = `Collection`
-const COMMENT = `Comment`
-const PRODUCT = `Product`
-const PRODUCT_OPTION = `ProductOption`
-const PRODUCT_VARIANT = `ProductVariant`
-const SHOP_POLICY = `ShopPolicy`
-const PRODUCT_TYPE = `ProductType`
-const PAGE = `Page`
+export const ARTICLE = `Article`
```
[the remark this answers is not in the document]
Comment (sidharthachatterjee) [reply packages/gatsby-source-shopify/src/nodes.js]: Done

**Comment by sidharthachatterjee** [comment]
Published in `gatsby-source-shopify@2.2.2`

**Comment by martingtheodo** [comment]
@sidharthachatterjee @freiksenet Thank you for reviewing the PR and for the warm welcome !

**Pull request merged**